deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -246,7 +247,11 @@ public interface RequestEncodingAdapter<REQUEST> {
     }
 
     public interface RequestDecodingAdapter<REQUEST> {
+        @Deprecated
         Optional<String> getFirstHeader(REQUEST request, String headerName);
+
+        @Nullable
+        String maybeFirstHeader(REQUEST request, String headerName);
```

Comment:
I think we need to make this a default method initially to avoid breaking existing implementations

Reply:
ah right you are